### PR TITLE
style: Add infrastructure to have a standalone author stylesheet collection

### PR DESCRIPTION
**Pull request by emilio**
Right now Gecko uses a whole `Stylist` for stuff like XBL / Shadow DOM.

That's not great, because it has tons of unrelated logic, and also eats up a lot of memory. Also, it prevents us to optimize style changes in shadow hosts the same way we do for the document.

These patches mostly rejigger stuff around so that you can define a `DocumentStylesheetSet` and then an `AuthorStylesheetSet`, which would contain just the Shadow DOM sheets / XBL resource sheets.

It still doesn't introduce any use for the later, but that will come later.

There's a patch in this PR that requires Gecko changes, posted in https://bugzilla.mozilla.org/show_bug.cgi?id=1436798.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20004)

### CODE REVIEW

File: components/style/stylesheet_set.rs [hunk 1]
```diff
@@ -175,11 +175,14 @@ where
         debug_assert_eq!(origin, Origin::UserAgent);
 
         // We could iterate over `origin_sheets(origin)` to ensure state is
-        // consistent (that the `dirty` member of the Entry is reset to
-        // `false`).
+        // consistent (that the `committed` member of the Entry is reset to
+        // `true`).
```
Comment (nox) [inline components/style/stylesheet_set.rs]: Not sure about that comment change.

File: ports/geckolib/glue.rs [hunk 2]
```diff
@@ -1252,7 +1253,13 @@ pub extern "C" fn Servo_StyleSet_NoteStyleSheetsChanged(
 ) {
     let mut data = PerDocumentStyleData::from_ffi(raw_data).borrow_mut();
     data.stylist.force_stylesheet_origins_dirty(OriginSet::from(changed_origins));
-    data.stylist.set_author_style_disabled(author_style_disabled);
+    let enabled =
+        if author_style_disabled {
```
Comment (nox) [inline ports/geckolib/glue.rs]: Could that change be propagated to wherever `author_style_disabled` come from?

**Review by nox — APPROVED** [review]
r=me with green geckotry

**Comment by emilio** [comment]
@bors-servo p=1

 * Coordiland

**Comment by emilio** [comment]
Whoops, forgot a commit

**Comment by emilio** [comment]
Err, to update bindings i mean

**Pull request merged**